x-pack/legacy/plugins/siem/public/containers/detection_engine/signals/api.test.ts

```diff
@@ -0,0 +1,291 @@
+    beforeEach(() => {
+      mockKibanaServices.mockClear();
+      fetchMock.mockClear();
+      fetchMock.mockImplementation(() => ({
+        response: {
+          ok: true,
+          message: 'success',
+          text: 'success',
+        },
+        body: {},
+      }));
+    });
+    test('check parameter url, body when closing a signal', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await updateSignalStatus({
+        query: mockStatusSignalQuery,
+        signal: abortCtrl.signal,
+        status: 'closed',
+      });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/signals/status', {
+        asResponse: true,
+        body:
+          '{"status":"closed","bool":{"filter":{"terms":{"_id":["b4ee5c32e3a321057edcc953ca17228c6fdfe5ba43fdbbdaffa8cefa11605cc5"]}}}}',
+        method: 'POST',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('check parameter url, body when opening a signal', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await updateSignalStatus({
+        query: mockStatusSignalQuery,
+        signal: abortCtrl.signal,
+        status: 'open',
+      });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/signals/status', {
+        asResponse: true,
+        body:
+          '{"status":"open","bool":{"filter":{"terms":{"_id":["b4ee5c32e3a321057edcc953ca17228c6fdfe5ba43fdbbdaffa8cefa11605cc5"]}}}}',
+        method: 'POST',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('happy path', async () => {
+      mockfetchSuccess({});
+      const signalsResp = await updateSignalStatus({
+        query: mockStatusSignalQuery,
+        signal: abortCtrl.signal,
+        status: 'open',
+      });
+      expect(signalsResp).toEqual({});
+    });
+    test('unhappy path', async () => {
+      mockfetchError();
+      try {
+        await updateSignalStatus({
+          query: mockStatusSignalQuery,
+          signal: abortCtrl.signal,
+          status: 'open',
+        });
+      } catch (exp) {
+        expect(exp).toBeInstanceOf(ToasterErrors);
+        expect(exp.message).toEqual('super mega error, it is not that bad');
+      }
+    });
+  });
+
+  describe('getSignalIndex', () => {
+    beforeEach(() => {
+      mockKibanaServices.mockClear();
+      fetchMock.mockClear();
+      fetchMock.mockImplementation(() => mockSignalIndex);
+    });
+    test('check parameter url', async () => {
+      mockfetchSuccess(null, fetchMock);
+
+      await getSignalIndex({ signal: abortCtrl.signal });
+      expect(fetchMock).toHaveBeenCalledWith('/api/detection_engine/index', {
+        method: 'GET',
+        signal: abortCtrl.signal,
+      });
+    });
+    test('happy path', async () => {
+      mockfetchSuccess(null, fetchMock);
+      const signalsResp = await getSignalIndex({
+        signal: abortCtrl.signal,
+      });
+      expect(signalsResp).toEqual(mockSignalIndex);
+    });
+    test('unhappy path', async () => {
```

Comment:
This one is a true false positive: we're expecting a bad response to cause an error to be thrown (which our actual `fetch` call may do), but it is not.